src/Microsoft.DotNet.Wpf/src/Shared/System/IO/FileHelper.cs

```diff
@@ -168,6 +168,11 @@ static internal void DeleteTemporaryFile(string filePath)
                     // We may not be able to delete the file if it's being used by some other process (e.g. Anti-virus check).
                     // There's nothing we can do in that case, so just eat the exception and leave the file behind
                 }
+                catch(System.UnauthorizedAccessException)
```

Comment:
Adding a new catch here can possibly change the code flow at https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/Speller.cs#L380. 

Although I don't have an example to substantiate my claim, but I would suggest checking this code flow for its expected behavior with these changes.

Reply:
Yes that is correct, mentioned in the Customer Impact. However, if you look into the .NET Framework code, it first demands `FileIOPermissionAccess.AllAccess`. If that fails, the `SecurityException` is silently caught and the execution continues.

Removing this check in .NET Core possibly introduced a regression in that not having rights to delete the file now crashes the process. This PR fixes would fix such regression. I will update the PR text accordingly. Thanks!

Reply:
Ok... I think in this case we can modify `UnloadDictionary` function to behave similar like in .NET Framework so that the code flow at https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/Speller.cs#L380 remains as it is.

Let me know if I am missing any point here or my suggestion won't work for you.

Reply:
Sorry I am not quite sure I understand - what change to the `UnloadDictionary` are you proposing?

Reply:
I am suggesting to make `UnloadDictionary` function same as it is in .NET Farmework i.e. wrapping the `FileHelper.DeleteTemporaryFile(filePath);` inside a try-catch block and suppressing the exception there instead of suppressing the exception here in `DeleteTemporaryFile`

Reply:
That would only fix the case for the dictionary though, the point is to avoid crashes when any temporary file cannot be deleted (and the motivation was crash due to cursor files). What is the issue with catching it inside `DeleteTemporaryFile`?